src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -393,6 +393,9 @@ private void deleteJob(String id) {
 			boolean jobDeleted = jobsToDelete.delete();
 			logger.debug(String.format("Job deleted for: %s - %b", id, jobDeleted));
 		}
+		else {
```

Comment:
I think by doing the empty check first we can simplify this a bit (ditch the else) such as:
```java
if (podsToDelete == null || ObjectUtils.isEmpty(podsToDelete.list().getItems())) {
  throw new IllegalStateException(String.format("Cannot delete Pod for task \"%s\" (reason: pod does not exist)", id));
}

logger.debug(String.format("Deleting Pod for task: %s", id));
boolean podsDeleted = podsToDelete.delete();
logger.debug(String.format("Pod deleted for: %s - %b", id, podsDeleted));
```

Wdyt?